### PR TITLE
[Doc] Update LightningTrainer MNIST example.

### PR DESCRIPTION
**Pull request by woshiyyya**
## Why are these changes needed?

Some users reported that there are some updates in lightning 2.0 and torchmetrics that our example are not compatible with. Update our example accordingly.

Changes:
- Update `Accuracy()` to `Accuracy(task=, num_classes=)`, the former one is deprecated in torchmetrics>0.9
- Update `evaluation_epoch_end()` to `on_evaluation_epoch_end()`, since the former is deprecated in ptl 2.0

The two changes above are backward compatible, so it won't affect the users using lightning/torchmetrics with an older version.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/train/examples/lightning/lightning_mnist_example.ipynb [hunk 1]
```diff
@@ -123,7 +110,9 @@
     "            nn.ReLU(),\n",
     "        )\n",
     "        self.lr = lr\n",
-    "        self.accuracy = Accuracy()\n",
+    "        self.accuracy = Accuracy(task=\"multiclass\", num_classes=10)\n",
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_mnist_example.ipynb:113]: CAn you include the package installation instructions at the top of this example?

File: doc/source/train/examples/lightning/lightning_mnist_example.ipynb [hunk 2]
```diff
@@ -154,11 +145,13 @@
     "        acc = self.accuracy(logits, y)\n",
     "        return loss, acc\n",
     "\n",
-    "    def validation_epoch_end(self, outputs):\n",
-    "        avg_loss = torch.stack([x[\"val_loss\"] for x in outputs]).mean()\n",
-    "        avg_acc = torch.stack([x[\"val_accuracy\"] for x in outputs]).mean()\n",
+    "    def on_validation_epoch_end(self):\n",
+    "        avg_loss = torch.stack(self.eval_loss).mean()\n",
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_mnist_example.ipynb:149]: nit: Can you explain in the PR description why these changes are needed (PTL 2.0) and whether these changes are backwards compatible with older versions?

File: doc/source/train/examples/lightning/lightning_mnist_example.ipynb [hunk 3]
```diff
@@ -259,22 +252,29 @@
...
+    "lightning_config = build_lightning_config_from_existing_code(use_gpu=use_gpu)\n",
+    "\n",
+    "if use_gpu:\n",
+    "    scaling_config = ScalingConfig(\n",
+    "        num_workers=4, use_gpu=True, resources_per_worker={\"CPU\": 1, \"GPU\": 1}\n",
+    "    )\n",
+    "else:\n",
+    "    scaling_config = ScalingConfig(\n",
+    "        num_workers=4, use_gpu=False, resources_per_worker={\"CPU\": 1}\n",
+    "    )\n",
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_mnist_example.ipynb:277]: Change this to something like:
```python
scaling_config = ScalingConfig(num_workers=4, use_gpu=use_gpu)
```

Since the `resources_per_worker` should be automatically set.

File: doc/source/train/examples/lightning/lightning_mnist_example.ipynb [hunk 4]
```diff
@@ -232,7 +225,7 @@
     "    # )\n",
     "    config_builder.trainer(\n",
     "        max_epochs=10,\n",
-    "        accelerator=\"cpu\",\n",
+    "        accelerator=\"gpu\" if use_gpu else \"cpu\",\n",
```
Comment (matthewdeng) [inline doc/source/train/examples/lightning/lightning_mnist_example.ipynb:228]: Should the user even need to set this? Can this be inferred and set from the `ScalingConfig`, and if not can we just raise a warning?

**Review by matthewdeng — APPROVED** [review]

File: doc/source/train/examples/lightning/lightning_mnist_example.ipynb (hunk 2 above)
Comment (woshiyyya) [reply doc/source/train/examples/lightning/lightning_mnist_example.ipynb:158]: Done
